### PR TITLE
BOLT 2,4: allow an error for HTLCs which expire too far away.

### PR DESCRIPTION
**Pull request by rustyrussell**

### CODE REVIEW

**Comment by pm47** [comment]
I wonder if this error should have the UPDATE flag as well? Even if the `channel_update` message doesn't have information on the maximum allowed expiry, we do provide it in `incorrect_cltv_expiry` error where it is not more helpful.

**Comment by rustyrussell** [comment]
Well, incorrect_cltv_expiry does help, since we might have changed it for the outgoing channel?  I guess we could have a max_cltv field in future updates, though I think this corner case isn't worth the bytes.

**Comment by pm47** [comment]
> Well, incorrect_cltv_expiry does help, since we might have changed it for the outgoing channel?

Oops, you're right!

**Comment by pm47** [comment]
Linking to #261

File: 02-peer-protocol.md [hunk 1]
```diff
@@ -519,6 +521,9 @@ In general, we need to complete one side of the exchange before dealing with the
 Fulfilling an HTLC is different: knowledge of the preimage is by definition irrevocable, 
 so we should fulfill the incoming HTLC as soon as we can to reduce latency.
 
+An HTLC with an extremely long expiry is a denial-of-service vector,
+so is not allowed: the exact value of reasonable is currently unclear
```
Comment (cdecker) [inline 02-peer-protocol.md]: "so it is"

Comment (rustyrussell) [reply 02-peer-protocol.md]: It's actually correct in British English, but slightly old-fashioned/formal. I often get picked up on it by Americans, for whom it's not idiomatic.  Will fix...

**Pull request merged**